src/hooks/useDimensions.tsx

```diff
@@ -0,0 +1,20 @@
+import { RefObject, useMemo, useSyncExternalStore } from 'react';
+
+function subscribeToResize(callback: () => void) {
+  window.addEventListener('resize', callback);
+
+  return () => {
+    window.removeEventListener('resize', callback);
+  };
+}
+
+export function useDimensions(ref: RefObject<HTMLElement>) {
```

Comment:
- The file name extension should be .ts and not .tsx

- This needs to go to common/hook/usedimensions.ts